diff -Nur neutrino-ddt.git/acinclude.m4 neutrino-ddt.git.patched/acinclude.m4
--- neutrino-ddt.git/acinclude.m4	2023-04-08 22:32:29.555304455 +0200
+++ neutrino-ddt.git.patched/acinclude.m4	2023-05-03 16:56:12.784508685 +0200
@@ -276,7 +276,7 @@
 AC_ARG_WITH(boxtype,
 	AS_HELP_STRING([--with-boxtype], [valid values: spark, generic, armbox, duckbox, spark7162, mipsbox]),
 	[case "${withval}" in
-		generic|armbox)
+		generic|armbox|mipsbox)
 			BOXTYPE="$withval"
 		;;
 		spark|spark7162)
