diff -Nur neutrino-ddt.git/src/driver/simple_display.cpp neutrino-ddt.git.patched/src/driver/simple_display.cpp
--- neutrino-ddt.git/src/driver/simple_display.cpp	2023-04-08 22:32:29.559304450 +0200
+++ neutrino-ddt.git.patched/src/driver/simple_display.cpp	2023-05-03 21:28:24.803415000 +0200
@@ -79,8 +79,49 @@
 static inline int dev_open()
 {
 	int fd = open(DISPLAY_DEV, O_RDWR);
-	if (fd < 0)
-		fprintf(stderr, "[neutrino] simple_display: open " DISPLAY_DEV ": %m\n");
+	
+	//
+	if(fd < 0)
+	{
+		// probe /dev/vfd
+		fd = open("/dev/vfd", O_RDWR);
+		
+		if(fd < 0)
+		{
+			// probe /dev/display
+			fd = open("/dev/display", O_RDWR);
+			
+			if(fd < 0)
+			{
+				// probe /dev/mcu
+				fd = open("/dev/mcu", O_RDWR);
+
+				if(fd < 0)
+				{
+					// probe /proc/vfd
+					fd = open("/proc/vfd", O_RDWR);
+				
+					if(fd < 0)
+					{
+						// probe /dev/dbox/fp
+						fd = open("/dev/dbox/fp", O_RDWR);
+		
+						if(fd < 0) 
+						{
+							// probe /dev/oled0
+							fd = open("/dev/oled0", O_RDWR);
+						
+							if(fd < 0)
+							{
+								fprintf(stderr, "[neutrino] simple_display: open " DISPLAY_DEV ": %m\n");
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	
 	return fd;
 }
 
diff -Nur neutrino-ddt.git/src/zapit/src/bouquets.cpp neutrino-ddt.git.patched/src/zapit/src/bouquets.cpp
--- neutrino-ddt.git/src/zapit/src/bouquets.cpp	2023-05-04 11:52:06.107266630 +0200
+++ neutrino-ddt.git.patched/src/zapit/src/bouquets.cpp	2023-05-03 22:56:32.453281000 +0200
@@ -800,7 +800,7 @@
 
 			if (strcasecmp("xml", extension.c_str()) == 0)
 				xml = true;
-			if ((strcasecmp("m3u", extension.c_str()) == 0) || (strcasecmp("m3u8", extension.c_str()) == 0))
+			if ( (strcasecmp("m3u", extension.c_str()) == 0) || (strcasecmp("m3u8", extension.c_str()) == 0) )
 				m3u = true;
 			if (strcasecmp("tv", extension.c_str()) == 0)
 				e2tv = true;
