diff -Nur neutrino-ddt.git/acinclude.m4 neutrino-ddt.git.patched/acinclude.m4
--- neutrino-ddt.git/acinclude.m4	2023-04-08 22:32:29.555304455 +0200
+++ neutrino-ddt.git.patched/acinclude.m4	2023-05-03 16:56:12.784508685 +0200
@@ -276,7 +276,7 @@
 AC_ARG_WITH(boxtype,
 	AS_HELP_STRING([--with-boxtype], [valid values: spark, generic, armbox, duckbox, spark7162, mipsbox]),
 	[case "${withval}" in
-		generic|armbox)
+		generic|armbox|mipsbox)
 			BOXTYPE="$withval"
 		;;
 		spark|spark7162)
diff -Nur neutrino-ddt.git/src/driver/simple_display.cpp neutrino-ddt.git.patched/src/driver/simple_display.cpp
--- neutrino-ddt.git/src/driver/simple_display.cpp	2023-04-08 22:32:29.559304450 +0200
+++ neutrino-ddt.git.patched/src/driver/simple_display.cpp	2023-05-03 21:28:24.803415358 +0200
@@ -79,8 +79,49 @@
 static inline int dev_open()
 {
 	int fd = open(DISPLAY_DEV, O_RDWR);
-	if (fd < 0)
-		fprintf(stderr, "[neutrino] simple_display: open " DISPLAY_DEV ": %m\n");
+	
+	//
+	if(fd < 0)
+	{
+		// probe /dev/vfd
+		fd = open("/dev/vfd", O_RDWR);
+		
+		if(fd < 0)
+		{
+			// probe /dev/display
+			fd = open("/dev/display", O_RDWR);
+			
+			if(fd < 0)
+			{
+				// probe /dev/mcu
+				fd = open("/dev/mcu", O_RDWR);
+
+				if(fd < 0)
+				{
+					// probe /proc/vfd
+					fd = open("/proc/vfd", O_RDWR);
+				
+					if(fd < 0)
+					{
+						// probe /dev/dbox/fp
+						fd = open("/dev/dbox/fp", O_RDWR);
+		
+						if(fd < 0) 
+						{
+							// probe /dev/oled0
+							fd = open("/dev/oled0", O_RDWR);
+						
+							if(fd < 0)
+							{
+								fprintf(stderr, "[neutrino] simple_display: open " DISPLAY_DEV ": %m\n");
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	
 	return fd;
 }
 
diff -Nur neutrino-ddt.git/src/gui/webtv_setup.cpp neutrino-ddt.git.patched/src/gui/webtv_setup.cpp
--- neutrino-ddt.git/src/gui/webtv_setup.cpp	2022-10-22 12:24:07.692823091 +0200
+++ neutrino-ddt.git.patched/src/gui/webtv_setup.cpp	2023-05-03 22:41:49.893077975 +0200
@@ -96,6 +96,7 @@
 			fileFilter.addFilter("xml");
 			fileFilter.addFilter("tv");
 			fileFilter.addFilter("m3u");
+			fileFilter.addFilter("m3u8");
 			fileBrowser.Filter = &fileFilter;
 
 			dirname = dirname.substr(0, dirname.rfind('/'));
diff -Nur neutrino-ddt.git/src/zapit/src/bouquets.cpp neutrino-ddt.git.patched/src/zapit/src/bouquets.cpp
--- neutrino-ddt.git/src/zapit/src/bouquets.cpp	2023-04-08 22:32:29.567304444 +0200
+++ neutrino-ddt.git.patched/src/zapit/src/bouquets.cpp	2023-05-03 22:56:32.453281490 +0200
@@ -800,7 +800,7 @@
 
 			if (strcasecmp("xml", extension.c_str()) == 0)
 				xml = true;
-			if (strcasecmp("m3u", extension.c_str()) == 0)
+			if ( (strcasecmp("m3u", extension.c_str()) == 0) || (strcasecmp("m3u8", extension.c_str()) == 0) )
 				m3u = true;
 			if (strcasecmp("tv", extension.c_str()) == 0)
 				e2tv = true;
@@ -904,6 +904,10 @@
 					{
 						epg_url = "";
 						epg_url = ReadMarkerValue(strLine, M3U_START_EPG_MARKER);
+						if (epg_url.empty())
+						{
+							epg_url = ReadMarkerValue(strLine, "x-tvg-url");
+						}
 						printf("tvg-url: %s\n", epg_url.c_str());
 						if (!epg_url.empty())
 						{
