diff -Nur titan.svn/autogen.sh titan.svn.patched/autogen.sh
--- titan.svn/autogen.sh	2022-06-27 22:22:53.913965000 +0200
+++ titan.svn.patched/autogen.sh	2023-05-07 09:08:09.633931630 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-package="tuxbox-neutrino"
+package="tuxbox-titan"
 
 srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
diff -Nur titan.svn/configure.ac titan.svn.patched/configure.ac
--- titan.svn/configure.ac	2022-06-27 22:22:53.913965000 +0200
+++ titan.svn.patched/configure.ac	2023-05-06 14:42:09.895452103 +0200
@@ -17,6 +17,7 @@
 AM_CONDITIONAL([ENABLE_MULTICOM324], [test x$enable_multicom324 = xtrue])
 AC_SUBST(MULTICOM324)
 
+# eplayer3
 AC_ARG_ENABLE(eplayer3,
 	AS_HELP_STRING(--enable-eplayer3,    enable eplayer3 support),
 	[case "${enableval}" in
@@ -25,6 +26,22 @@
 AM_CONDITIONAL(ENABLE_EPLAYER3,test "$enable_eplayer3" = "yes")
 AC_SUBST(EPLAYER3)
 
+# gstreamer
+AC_ARG_ENABLE(gstreamer,
+	AS_HELP_STRING(--enable-gstreamer,include gstreamer as player engine support),
+	,[enable_gstreamer=no])
+AM_CONDITIONAL(ENABLE_EPLAYER4,test "$enable_gstreamer" = "yes")
+
+if test "$enable_gstreamer" = "yes"; then
+AC_DEFINE(EPLAYER4,1,[include gstreamer as player engine support])
+
+AC_ARG_WITH(gstversion,
+	AS_HELP_STRING([--with-gstversion],[use gstreamer version (major.minor)]),
+	[GST_MAJORMINOR=$withval],[GST_MAJORMINOR=1.0])
+
+PKG_CHECK_MODULES(GSTREAMER, gstreamer-$GST_MAJORMINOR gstreamer-pbutils-$GST_MAJORMINOR)
+fi
+
 AC_MSG_RESULT([
 summary:
 ----------------------------------------
diff -Nur titan.svn/Makefile.am titan.svn.patched/Makefile.am
--- titan.svn/Makefile.am	2022-06-27 22:22:53.917965000 +0200
+++ titan.svn.patched/Makefile.am	2023-05-06 14:32:09.001581995 +0200
@@ -1,3 +1,3 @@
 AUTOMAKE_OPTIONS = gnu
 
-SUBDIRS = titan plugins
\ No newline at end of file
+SUBDIRS = titan
diff -Nur titan.svn/plugins/browser/netsurf/Makefile.am titan.svn.patched/plugins/browser/netsurf/Makefile.am
--- titan.svn/plugins/browser/netsurf/Makefile.am	2022-06-27 22:22:40.490115000 +0200
+++ titan.svn.patched/plugins/browser/netsurf/Makefile.am	2023-05-07 22:43:33.918774991 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libbrowser.la
+noinst_LTLIBRARIES = libbrowser.la
 libbrowser_la_SOURCES = browser.c
 
diff -Nur titan.svn/plugins/games/catcatch/Makefile.am titan.svn.patched/plugins/games/catcatch/Makefile.am
--- titan.svn/plugins/games/catcatch/Makefile.am	2022-06-27 22:22:40.490115000 +0200
+++ titan.svn.patched/plugins/games/catcatch/Makefile.am	2023-05-07 22:44:43.479288879 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libcatcatch.la
+noinst_LTLIBRARIES = libcatcatch.la
 libcatcatch_la_SOURCES = catcatch.c
 
diff -Nur titan.svn/plugins/games/sudoku/Makefile.am titan.svn.patched/plugins/games/sudoku/Makefile.am
--- titan.svn/plugins/games/sudoku/Makefile.am	2022-06-27 22:22:40.530114000 +0200
+++ titan.svn.patched/plugins/games/sudoku/Makefile.am	2023-05-07 22:45:03.740557464 +0200
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = libsudoku.la
+noinst_LTLIBRARIES = libsudoku.la
 libsudoku_la_SOURCES = sudoku.c
 
 bin_PROGRAMS = sudoku_core
diff -Nur titan.svn/plugins/games/wins3/Makefile.am titan.svn.patched/plugins/games/wins3/Makefile.am
--- titan.svn/plugins/games/wins3/Makefile.am	2022-06-27 22:22:40.534114000 +0200
+++ titan.svn.patched/plugins/games/wins3/Makefile.am	2023-05-07 22:45:28.310071764 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libwins3.la
+noinst_LTLIBRARIES = libwins3.la
 libwins3_la_SOURCES = wins3.c
 
diff -Nur titan.svn/plugins/gui/e2starter/Makefile.am titan.svn.patched/plugins/gui/e2starter/Makefile.am
--- titan.svn/plugins/gui/e2starter/Makefile.am	2022-06-27 22:22:40.546114000 +0200
+++ titan.svn.patched/plugins/gui/e2starter/Makefile.am	2023-05-07 22:47:51.854503106 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libe2starter.la
+noinst_LTLIBRARIES = libe2starter.la
 libe2starter_la_SOURCES = e2starter.c
 
diff -Nur titan.svn/plugins/gui/kodistarter/Makefile.am titan.svn.patched/plugins/gui/kodistarter/Makefile.am
--- titan.svn/plugins/gui/kodistarter/Makefile.am	2022-06-27 22:22:40.566114000 +0200
+++ titan.svn.patched/plugins/gui/kodistarter/Makefile.am	2023-05-07 22:48:22.232217069 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libkodistarter.la
+noinst_LTLIBRARIES = libkodistarter.la
 libkodistarter_la_SOURCES = kodistarter.c
 
diff -Nur titan.svn/plugins/infos/imdb/Makefile.am titan.svn.patched/plugins/infos/imdb/Makefile.am
--- titan.svn/plugins/infos/imdb/Makefile.am	2022-06-27 22:22:40.630113000 +0200
+++ titan.svn.patched/plugins/infos/imdb/Makefile.am	2023-05-07 22:48:41.617300930 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libimdb.la
+noinst_LTLIBRARIES = libimdb.la
 libimdb_la_SOURCES = imdb.c
 
diff -Nur titan.svn/plugins/infos/imdbapi/Makefile.am titan.svn.patched/plugins/infos/imdbapi/Makefile.am
--- titan.svn/plugins/infos/imdbapi/Makefile.am	2022-06-27 22:22:40.610114000 +0200
+++ titan.svn.patched/plugins/infos/imdbapi/Makefile.am	2023-05-07 22:48:55.610078901 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libimdbapi.la
+noinst_LTLIBRARIES = libimdbapi.la
 libimdbapi_la_SOURCES = imdbapi.c
 
diff -Nur titan.svn/plugins/infos/mboxinfo/Makefile.am titan.svn.patched/plugins/infos/mboxinfo/Makefile.am
--- titan.svn/plugins/infos/mboxinfo/Makefile.am	2022-06-27 22:22:40.906110000 +0200
+++ titan.svn.patched/plugins/infos/mboxinfo/Makefile.am	2023-05-07 22:49:07.642745116 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libmboxinfo.la
+noinst_LTLIBRARIES = libmboxinfo.la
 libmboxinfo_la_SOURCES = mboxinfo.c
 
diff -Nur titan.svn/plugins/infos/stock/Makefile.am titan.svn.patched/plugins/infos/stock/Makefile.am
--- titan.svn/plugins/infos/stock/Makefile.am	2022-06-27 22:22:40.706112000 +0200
+++ titan.svn.patched/plugins/infos/stock/Makefile.am	2023-05-07 22:49:23.731632110 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libstock.la
+noinst_LTLIBRARIES = libstock.la
 libstock_la_SOURCES = stock.c
 
diff -Nur titan.svn/plugins/infos/streaminfo/Makefile.am titan.svn.patched/plugins/infos/streaminfo/Makefile.am
--- titan.svn/plugins/infos/streaminfo/Makefile.am	2022-06-27 22:22:40.698113000 +0200
+++ titan.svn.patched/plugins/infos/streaminfo/Makefile.am	2023-05-07 22:49:37.756401954 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libstreaminfo.la
+noinst_LTLIBRARIES = libstreaminfo.la
 libstreaminfo_la_SOURCES = streaminfo.c
 
diff -Nur titan.svn/plugins/infos/tinews/Makefile.am titan.svn.patched/plugins/infos/tinews/Makefile.am
--- titan.svn/plugins/infos/tinews/Makefile.am	2022-06-27 22:22:40.662113000 +0200
+++ titan.svn.patched/plugins/infos/tinews/Makefile.am	2023-05-07 22:49:50.189081929 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libtinews.la
+noinst_LTLIBRARIES = libtinews.la
 libtinews_la_SOURCES = tinews.c
 
diff -Nur titan.svn/plugins/infos/tmdb/Makefile.am titan.svn.patched/plugins/infos/tmdb/Makefile.am
--- titan.svn/plugins/infos/tmdb/Makefile.am	2022-06-27 22:22:40.702113000 +0200
+++ titan.svn.patched/plugins/infos/tmdb/Makefile.am	2023-05-07 22:50:05.109895077 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libtmdb.la
+noinst_LTLIBRARIES = libtmdb.la
 libtmdb_la_SOURCES = tmdb.c
 
diff -Nur titan.svn/plugins/infos/weather/Makefile.am titan.svn.patched/plugins/infos/weather/Makefile.am
--- titan.svn/plugins/infos/weather/Makefile.am	2022-06-27 22:22:40.726112000 +0200
+++ titan.svn.patched/plugins/infos/weather/Makefile.am	2023-05-07 22:50:21.790800598 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libweather.la
+noinst_LTLIBRARIES = libweather.la
 libweather_la_SOURCES = weather.c
 
diff -Nur titan.svn/plugins/infos/wm2018/Makefile.am titan.svn.patched/plugins/infos/wm2018/Makefile.am
--- titan.svn/plugins/infos/wm2018/Makefile.am	2022-06-27 22:22:40.894110000 +0200
+++ titan.svn.patched/plugins/infos/wm2018/Makefile.am	2023-05-07 22:50:36.191579516 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libwm2018.la
+noinst_LTLIBRARIES = libwm2018.la
 libwm2018_la_SOURCES = wm2018.c
 
diff -Nur titan.svn/plugins/network/dlna/Makefile.am titan.svn.patched/plugins/network/dlna/Makefile.am
--- titan.svn/plugins/network/dlna/Makefile.am	2022-06-27 22:22:41.210107000 +0200
+++ titan.svn.patched/plugins/network/dlna/Makefile.am	2023-05-07 22:50:55.196603699 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libdlna.la
+noinst_LTLIBRARIES = libdlna.la
 libdlna_la_SOURCES = dlna.c
 
diff -Nur titan.svn/plugins/network/facebook/Makefile.am titan.svn.patched/plugins/network/facebook/Makefile.am
--- titan.svn/plugins/network/facebook/Makefile.am	2022-06-27 22:22:40.914110000 +0200
+++ titan.svn.patched/plugins/network/facebook/Makefile.am	2023-05-07 22:51:09.033346828 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libfacebook.la
+noinst_LTLIBRARIES = libfacebook.la
 libfacebook_la_SOURCES = facebook.c
 
diff -Nur titan.svn/plugins/network/ipsec/Makefile.am titan.svn.patched/plugins/network/ipsec/Makefile.am
--- titan.svn/plugins/network/ipsec/Makefile.am	2022-06-27 22:22:40.962110000 +0200
+++ titan.svn.patched/plugins/network/ipsec/Makefile.am	2023-05-07 22:51:22.746081305 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libipsec.la
+noinst_LTLIBRARIES = libipsec.la
 libipsec_la_SOURCES = ipsec.c
 
diff -Nur titan.svn/plugins/network/minisatip/Makefile.am titan.svn.patched/plugins/network/minisatip/Makefile.am
--- titan.svn/plugins/network/minisatip/Makefile.am	2022-06-27 22:22:40.974109000 +0200
+++ titan.svn.patched/plugins/network/minisatip/Makefile.am	2023-05-07 22:51:34.754722970 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libminisatip.la
+noinst_LTLIBRARIES = libminisatip.la
 libminisatip_la_SOURCES = minisatip.c
 
diff -Nur titan.svn/plugins/network/networkbrowser/Makefile.am titan.svn.patched/plugins/network/networkbrowser/Makefile.am
--- titan.svn/plugins/network/networkbrowser/Makefile.am	2022-06-27 22:22:41.062108000 +0200
+++ titan.svn.patched/plugins/network/networkbrowser/Makefile.am	2023-05-07 22:51:52.919691010 +0200
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = libnetworkbrowser.la
+noinst_LTLIBRARIES = libnetworkbrowser.la
 libnetworkbrowser_la_SOURCES = networkbrowser.c
 libnetworkbrowser_la_LIBADD = netlib/libnetscan.la
 
diff -Nur titan.svn/plugins/network/satipclient/Makefile.am titan.svn.patched/plugins/network/satipclient/Makefile.am
--- titan.svn/plugins/network/satipclient/Makefile.am	2022-06-27 22:22:41.058108000 +0200
+++ titan.svn.patched/plugins/network/satipclient/Makefile.am	2023-05-07 22:52:30.381678421 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libsatipclient.la
+noinst_LTLIBRARIES = libsatipclient.la
 libsatipclient_la_SOURCES = satipclient.c
 
diff -Nur titan.svn/plugins/network/xupnpd/Makefile.am titan.svn.patched/plugins/network/xupnpd/Makefile.am
--- titan.svn/plugins/network/xupnpd/Makefile.am	2022-06-27 22:22:41.082108000 +0200
+++ titan.svn.patched/plugins/network/xupnpd/Makefile.am	2023-05-07 22:53:30.536847917 +0200
@@ -1,3 +1,3 @@
-lib_LTLIBRARIES = libxupnpd.la
+noinst_LTLIBRARIES = libxupnpd.la
 libxupnpd_la_SOURCES = xupnpd.c
 
diff -Nur titan.svn/plugins/tools/autotimer/autotimer.c titan.svn.patched/plugins/tools/autotimer/autotimer.c
--- titan.svn/plugins/tools/autotimer/autotimer.c	2022-06-27 22:22:41.198107000 +0200
+++ titan.svn.patched/plugins/tools/autotimer/autotimer.c	2023-05-07 20:29:20.465352252 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 char pluginname[] = "Auto Timer";
 char plugindesc[] = "Extensions";
diff -Nur titan.svn/plugins/tools/callmonitor1/callmonitor1.c titan.svn.patched/plugins/tools/callmonitor1/callmonitor1.c
--- titan.svn/plugins/tools/callmonitor1/callmonitor1.c	2022-06-27 22:22:41.198107000 +0200
+++ titan.svn.patched/plugins/tools/callmonitor1/callmonitor1.c	2023-05-07 20:39:40.927772900 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 #include "callmon_main.h"
 
 
diff -Nur titan.svn/plugins/tools/instar/instar.c titan.svn.patched/plugins/tools/instar/instar.c
--- titan.svn/plugins/tools/instar/instar.c	2022-06-27 22:22:41.250106000 +0200
+++ titan.svn.patched/plugins/tools/instar/instar.c	2023-05-07 20:40:14.407472020 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 #include "instar.h"
 
diff -Nur titan.svn/plugins/tools/keylock/keylock.c titan.svn.patched/plugins/tools/keylock/keylock.c
--- titan.svn/plugins/tools/keylock/keylock.c	2022-06-27 22:22:41.278106000 +0200
+++ titan.svn.patched/plugins/tools/keylock/keylock.c	2023-05-07 20:40:31.471318690 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 char pluginname[] = "KeyLock";
 char plugindesc[] = "Extensions";
diff -Nur titan.svn/plugins/tools/lcdpearl1/lcdpearl1.c titan.svn.patched/plugins/tools/lcdpearl1/lcdpearl1.c
--- titan.svn/plugins/tools/lcdpearl1/lcdpearl1.c	2023-05-04 21:46:29.004212188 +0200
+++ titan.svn.patched/plugins/tools/lcdpearl1/lcdpearl1.c	2023-05-07 20:41:00.375058971 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 #include "lcd_weather.h"
 
diff -Nur titan.svn/plugins/tools/lcdsamsung/lcdsamsung.c titan.svn.patched/plugins/tools/lcdsamsung/lcdsamsung.c
--- titan.svn/plugins/tools/lcdsamsung/lcdsamsung.c	2023-05-04 21:46:28.984212214 +0200
+++ titan.svn.patched/plugins/tools/lcdsamsung/lcdsamsung.c	2023-05-07 20:41:25.270835275 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 #include "lcd_weather.h"
 
@@ -533,7 +533,7 @@
 		hr = atoi(gettime(NULL, "%H"));
 		min = atoi(gettime(NULL, "%M"));
 		
-		// TV Programm läuft
+		// TV Programm l\E4uft
 		if(status.infobaraktiv == 1)
 		{
 			if(ostrcmp(getconfig("write_fb_to_jpg", NULL), "yes") == 0) {
diff -Nur titan.svn/plugins/tools/multiimage/MultiImage.c titan.svn.patched/plugins/tools/multiimage/MultiImage.c
--- titan.svn/plugins/tools/multiimage/MultiImage.c	2022-06-27 22:22:41.638102000 +0200
+++ titan.svn.patched/plugins/tools/multiimage/MultiImage.c	2023-05-07 20:41:44.430663124 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 char pluginname[] = "MultiImage";
 char plugindesc[] = "Start Images from USB device";
diff -Nur titan.svn/plugins/tools/permtime/permtime.c titan.svn.patched/plugins/tools/permtime/permtime.c
--- titan.svn/plugins/tools/permtime/permtime.c	2022-06-27 22:22:41.654102000 +0200
+++ titan.svn.patched/plugins/tools/permtime/permtime.c	2023-05-07 20:42:13.282403890 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 char pluginname[] = "Permanent Time";
 char plugindesc[] = "Extensions";
@@ -159,4 +159,4 @@
 		permtime_main();
 		
 	}
-}
\ No newline at end of file
+}
diff -Nur titan.svn/plugins/tools/sispmctl/sispmctl.c titan.svn.patched/plugins/tools/sispmctl/sispmctl.c
--- titan.svn/plugins/tools/sispmctl/sispmctl.c	2022-06-27 22:22:41.726101000 +0200
+++ titan.svn.patched/plugins/tools/sispmctl/sispmctl.c	2023-05-07 20:42:54.890030063 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 #include "sispmctl.h"
 
diff -Nur titan.svn/plugins/tools/stopifnotused/stopifnotused.c titan.svn.patched/plugins/tools/stopifnotused/stopifnotused.c
--- titan.svn/plugins/tools/stopifnotused/stopifnotused.c	2022-06-27 22:22:41.774100000 +0200
+++ titan.svn.patched/plugins/tools/stopifnotused/stopifnotused.c	2023-05-07 20:43:11.313882506 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 char pluginname[] = "Stop if not used";
 char plugindesc[] = "Auschalten der Box wenn in einem bestimmten Zeitraum keine Aktion erfolgte";
diff -Nur titan.svn/plugins/tools/tsschnitt/tsschnitt.c titan.svn.patched/plugins/tools/tsschnitt/tsschnitt.c
--- titan.svn/plugins/tools/tsschnitt/tsschnitt.c	2023-05-04 21:46:28.828212421 +0200
+++ titan.svn.patched/plugins/tools/tsschnitt/tsschnitt.c	2023-05-07 20:43:41.197614032 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 char pluginname[] = "Record cut";
 char plugindesc[] = "Extensions";
diff -Nur titan.svn/plugins/tools/wlandisplay/wlandisplay.c titan.svn.patched/plugins/tools/wlandisplay/wlandisplay.c
--- titan.svn/plugins/tools/wlandisplay/wlandisplay.c	2023-05-04 21:46:29.016212172 +0200
+++ titan.svn.patched/plugins/tools/wlandisplay/wlandisplay.c	2023-05-07 20:44:07.081381496 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 #include "lcd_weather.h"
 
@@ -528,7 +528,7 @@
 		hr = atoi(gettime(NULL, "%H"));
 		min = atoi(gettime(NULL, "%M"));
 		
-		// TV Programm läuft
+		// TV Programm l\E4uft
 		if(status.infobaraktiv == 1)
 		{
 			if(ostrcmp(getconfig("write_fb_to_jpg", NULL), "yes") == 0) {
diff -Nur titan.svn/plugins/tools/zapback/zapback.c titan.svn.patched/plugins/tools/zapback/zapback.c
--- titan.svn/plugins/tools/zapback/zapback.c	2022-06-27 22:22:41.978098000 +0200
+++ titan.svn.patched/plugins/tools/zapback/zapback.c	2023-05-07 20:44:22.373244119 +0200
@@ -1,6 +1,6 @@
-#include "../../titan/struct.h"
-#include "../../titan/debug.h"
-#include "../../titan/header.h"
+#include "../titan/struct.h"
+#include "../titan/debug.h"
+#include "../titan/header.h"
 
 char pluginname[] = "zapback (Werbezapper)";
 char plugindesc[] = "Extensions";
Binary files titan.svn/.svn/wc.db and titan.svn.patched/.svn/wc.db differ
diff -Nur titan.svn/titan/AUTHORS titan.svn.patched/titan/AUTHORS
--- titan.svn/titan/AUTHORS	1970-01-01 01:00:00.000000000 +0100
+++ titan.svn.patched/titan/AUTHORS	2023-05-08 10:18:40.877179470 +0200
@@ -0,0 +1 @@
+titan
diff -Nur titan.svn/titan/autogen.sh titan.svn.patched/titan/autogen.sh
--- titan.svn/titan/autogen.sh	2022-06-27 22:22:39.942121000 +0200
+++ titan.svn.patched/titan/autogen.sh	2023-05-08 01:40:01.444759000 +0200
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-package="tuxbox-neutrino"
+package="tuxbox-titan"
 
 srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
@@ -69,4 +69,4 @@
 echo "  autoheader"
 autoheader
 echo "  automake --add-missing"
-automake --add-missing 
\ No newline at end of file
+automake --add-missing 
diff -Nur titan.svn/titan/ChangeLog titan.svn.patched/titan/ChangeLog
--- titan.svn/titan/ChangeLog	1970-01-01 01:00:00.000000000 +0100
+++ titan.svn.patched/titan/ChangeLog	2023-05-08 10:18:25.457542222 +0200
@@ -0,0 +1 @@
+titan
diff -Nur titan.svn/titan/configure.ac titan.svn.patched/titan/configure.ac
--- titan.svn/titan/configure.ac	2022-06-27 22:22:39.938121000 +0200
+++ titan.svn.patched/titan/configure.ac	2023-05-08 01:46:39.179555000 +0200
@@ -25,6 +25,22 @@
 AM_CONDITIONAL(ENABLE_EPLAYER3,test "$enable_eplayer3" = "yes")
 AC_SUBST(EPLAYER3)
 
+# gstreamer
+AC_ARG_ENABLE(gstreamer,
+	AS_HELP_STRING(--enable-gstreamer,include gstreamer as player engine support),
+	,[enable_gstreamer=no])
+AM_CONDITIONAL(ENABLE_EPLAYER4,test "$enable_gstreamer" = "yes")
+
+if test "$enable_gstreamer" = "yes"; then
+AC_DEFINE(EPLAYER4,1,[include gstreamer as player engine support])
+
+AC_ARG_WITH(gstversion,
+	AS_HELP_STRING([--with-gstversion],[use gstreamer version (major.minor)]),
+	[GST_MAJORMINOR=$withval],[GST_MAJORMINOR=1.0])
+
+PKG_CHECK_MODULES(GSTREAMER, gstreamer-$GST_MAJORMINOR gstreamer-pbutils-$GST_MAJORMINOR)
+fi
+
 AC_MSG_RESULT([
 summary:
 ----------------------------------------
diff -Nur titan.svn/titan/header.h titan.svn.patched/titan/header.h
--- titan.svn/titan/header.h	2023-05-04 21:46:27.528214147 +0200
+++ titan.svn.patched/titan/header.h	2023-05-07 18:26:22.228958304 +0200
@@ -8,9 +8,10 @@
 #ifndef HEADER_H
 #define HEADER_H
 
+
 // libeplayer3/playback/playback.h
 int8_t PlaybackDieNow(int8_t val);
-#ifndef EPLAYER4 
+#ifdef EPLAYER3
 bool PlaybackDieNowRegisterCallback(PlaybackDieNowCallback callback);
 #endif
 // ca.h
diff -Nur titan.svn/titan/Makefile.am titan.svn.patched/titan/Makefile.am
--- titan.svn/titan/Makefile.am	2022-06-27 22:22:39.690124000 +0200
+++ titan.svn.patched/titan/Makefile.am	2023-05-08 10:24:03.105421303 +0200
@@ -1,18 +1,17 @@
-ALL_CFLAGS= -DSH4 -DEXTEPLAYER3 -DEPLAYER3 -DSH4 -DDVDPLAYER -Os -export-dynamic -Wall -Wno-unused-but-set-variable
-
-INCLUDES = \
-	-I../../../tufsbox/cdkroot/usr/include \
-	-I../../../tufsbox/cdkroot/usr/include/freetype2 \
-	-I../../../tufsbox/cdkroot/usr/include/openssl \
-	-I../../driver/bpamem \
-	-I../../../apps/tools/libeplayer3/include \
-	-I../../../apps/tools/libmme_image \
-	-Ilibdreamdvd \
-	-Ititan
+ALL_CFLAGS= -Os -export-dynamic -Wall -Wno-unused-but-set-variable
 
 bin_PROGRAMS = titan
 
 titan_SOURCES =	titan.c
 
-titan_LDADD = -lm -lpthread -ldl -lpng -lfreetype -leplayer3 -ldreamdvd -ljpeg -lz -lmmeimage -lmme_host -lssl -lcrypto
+titan_LDADD = -lm -lpthread -ldl -lpng -lfreetype -ldreamdvd -ljpeg -lz -lssl -lcrypto -lipkg
+
+if ENABLE_EPLAYER3
+titan_LDADD += -leplayer3 -lass
+endif
+
+if ENABLE_EPLAYER4
+titan_LDADD += @GSTREAMER_LIBS@
+endif
+
 titan_CFLAGS = $(ALL_CFLAGS)
diff -Nur titan.svn/titan/NEWS titan.svn.patched/titan/NEWS
--- titan.svn/titan/NEWS	1970-01-01 01:00:00.000000000 +0100
+++ titan.svn.patched/titan/NEWS	2023-05-08 10:18:56.716805863 +0200
@@ -0,0 +1 @@
+titan
diff -Nur titan.svn/titan/player.h titan.svn.patched/titan/player.h
--- titan.svn/titan/player.h	2023-05-04 21:46:27.528214147 +0200
+++ titan.svn.patched/titan/player.h	2023-05-07 15:12:45.539832732 +0200
@@ -3499,7 +3499,8 @@
 printf("return tracklist\n");
 		return TrackList;
 	#endif
-	#else
+	#endif
+	#ifdef EPLAYER4
 printf("return tracklist4b\n");
 
 		return TrackList4;
diff -Nur titan.svn/titan/README titan.svn.patched/titan/README
--- titan.svn/titan/README	1970-01-01 01:00:00.000000000 +0100
+++ titan.svn.patched/titan/README	2023-05-08 10:19:12.248438582 +0200
@@ -0,0 +1 @@
+titan
diff -Nur titan.svn/titan/titan.c titan.svn.patched/titan/titan.c
--- titan.svn/titan/titan.c	2023-05-04 21:46:27.520214158 +0200
+++ titan.svn.patched/titan/titan.c	2023-05-07 21:04:52.832048059 +0200
@@ -272,8 +272,10 @@
 #include "settings_overclocking.h"
 #include "settings_pluginbutton.h"
 #ifndef SIMULATE
+#ifdef MIPSEL
 #include "bcm.h"
 #endif
+#endif
 #include "settings_mediabutton.h"
 #include "oled.h"
 #ifdef MIPSEL
@@ -291,7 +293,7 @@
 #define ACCEL_DEBUG
 
 //gAccel *gAccel::instance;
-#define BCM_ACCEL
+//#define BCM_ACCEL
 #endif
 
 #ifdef HAVE_HISILICON_ACCEL 
